catalyst-gateway-crates/c509-certificate/src/tables.rs

```diff
@@ -0,0 +1,78 @@
+//! A bimap table for bidirectional lookup.
+
+use std::hash::Hash;
+
+use asn1_rs::Oid;
+use bimap::BiMap;
+
+/// A trait that represents a table structure with key-value pairs.
+///
+/// # Type Parameters
+///
+/// * `K` - The type of the keys in the table.
+/// * `V` - The type of the values in the table.
+pub(crate) trait TableTrait<K, V> {
```

Comment:
For what purpose we need this trait ?
Seems like it's not used anywhere actually, better remove it

Reply:
There are several tables in C509, in the future implementation, will consider removing this if it is not necessary

Reply:
But my point that at this moment we are not actually using this trait in any place of this crate.
I would rather remove it as we are not using it anywhere, if it will be needed, we'll add it in future.
Lets not to over design current implementation, because we dont know at this point of time how we want this generalisation of c509 tables, maybe even it would better to not use `trait` and implement everything just on enums. Such decisions I would defer at that time when we will actually need such generalisation.
If we dont have one right now, it is better to make it simple.